### DISCUSSION TITLE
Next.js sends back empty html from the server

### DISCUSSION THREAD

**Discussion by mohdashraf010897** [vercel/next.js#16389] [Help] [answered]
Next.js does client Side rendering although build status shows my pages to be prerendered at the server-side, as far as I could understand I should get some HTML in the response of next.js server instead I get a blank page(no HTML) and what happens is my whole app is client-side rendered ruining the whole purpose of getting next.js.

my expected behavior for my app should be that I should get HTML inside the page source but what I am actually getting is an empty div with id="__next" as a response with some scripts to generate the page using CSR; ruining the whole purpose of using next.js.

although the build shows that my particular page is a static one, seeing which I expect a pre-rendered HTML but even if I use getserversideprops or getstaticprops to enforce the behaviour, all in vain...
![image](https://user-images.githubusercontent.com/57627350/90756085-91daef80-e2f9-11ea-8cf9-5e481799ca28.png)

Page Source : 
![image](https://user-images.githubusercontent.com/57627350/90758418-a10f6c80-e2fc-11ea-97a8-e646e5f3b994.png)

Can someone tell me where exactly am I going wrong? 
Please excuse any stupidity if you find considering the fact that I am a complete beginner.

Thanks in advance.

**Comment by timneutkens** [comment, marked as answer]
Are you using redux-persist by any chance?

**Reply by mohdashraf010897** [reply]
YEs

**Reply by timneutkens** [reply]
redux-persist's provider opts out of pre-rendering unfortunately: https://github.com/vercel/next.js/discussions/16044

**Reply by mohdashraf010897** [reply]
And what is the solution to this?

**Reply by timneutkens** [reply]
You're using a library that is not compatible with pre-rendering so that's what causing this. Would recommend to not use redux-persist.

**Reply by rahulsuresh-git** [reply]
What is the alternative to this? Is there anyway to persist user actions without having to give up on SSG?

**Reply by mohdashraf010897** [reply]
Yes, just use localStorage manually on the client-side instead of relying on Redux-persist. Tim has got a point when he says not to use redux-persist since it relies on local storage which is not present at the server-side.

**Reply by itsmidhunnair** [reply]
can i get a verified answer?

**Reply by Deepanshu088** [reply]
@timneutkens  Thanks to your response, my search has ended after 2 days. Truly appreciated.

**Comment by mohdashraf010897** [comment]
Yes

**Comment by AdamDiament** [comment]
I am experiencing the same thing but I do not use redux-persist. I use apolloClient with prismic- should that be a problem?

**Reply by maisesports** [reply]
Did you find a solution for this? Not using redux-persist and having the same problem. I also use apolloClient.

**Reply by thugzook** [reply]
Any luck on this issue? Also using apolloClient... 2022

**Reply by mechadragon01** [reply]
Same

**Reply by tirrth** [reply]
Did anyone find a solution to the apolloClient issue?

**Comment by mohdashraf010897** [comment]
I dropped redux-persist from my project and it worked fine, also check for
anything that doesn't run on server-side; maybe that's causing an error.

**Comment by jfbaraky** [comment]
For now, I've created a conditional PersistGate that only run on browser:

```typescript
import React from 'react';
import { Persistor } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react';

type Props = {
  loading?: boolean;
  persistor: Persistor;
};

/**
 * SSR aware persist gate
 * @param props Same as the original PersistGate
 * @returns persisted or un-persisted component
 */
export const PersistGateSSR: React.FC<Props> = (props) => {
  if (!process.browser) {
    // Not on browser, nothing to persist here
    return <>{props.children}</>;
  } else {
    return (
      <PersistGate loading={props.loading} persistor={props.persistor}>
        {props.children}
      </PersistGate>
    );
  }
};

```

**Reply by rahimimo** [reply]
I tested the above solution @jfbaraky but my selectors cannot select anything from the store in the static site geneation. (when I disable javascript)

**Reply by jfbaraky** [reply]
The redux-persist is a client based solution for the localStorage. I don't think it should work without the client javascript.

**Reply by rahimimo** [reply]
I am using static site generation of nextjs, and when I disable javascript to check what is being generated and what google sees I get an empty blank page. 
https://github.com/rt2zz/redux-persist/issues/926

**Comment by Kasrabzm** [comment]
mine was `LocalizationProvider` from `@mui/lab/LocalizationProvider` that was causing the blank page to show

**Reply by emanuelavram** [reply]
did you replace the package with something else or what was your solution?

**Reply by huzaifa3115** [reply]
Replace with (import { LocalizationProvider } from '@mui/x-date-pickers')


![image](https://github.com/vercel/next.js/assets/38154499/2f15385c-3238-407d-852f-7ae2751d715c)

**Comment by IGoRFonin** [comment]
Try to use children as function.
```
<PersistGate loading={null} persistor={persistorRef.current as Persistor}>
    {() => children}
</PersistGate>
```

**Reply by garypotato** [reply]
Thanks mate, I think your answer should be the best answer. It works for me, but what's the reason behind?
